src/geoserver-create-imagemosaic-datastore/index.js

```diff
@@ -0,0 +1,138 @@
+ * @param {Array} inputs The inputs for this process
+ *   First input is the workspace to publish to
+ *   Second input is the name of the coverage store to be created
+ * @example
+    {
+       "id": 123,
+       "type": "geoserver-create-imagemosaic-datastore",
+       "inputs": [
+           "klips",
+           "my-coverageStore"
+        ]
+    }
+ */
+const geoserverCreateImageMosaicDatastore = async (workerJob, inputs) => {
+  const ws = inputs[0];
+  const covStore = inputs[1];
+  const geoServerAvailable = await isGeoServerAvailable();
+
+  if (!geoServerAvailable) {
+    console.log('Geoserver not available');
+    console.log('Job should be requeued!');
+    workerJob.missingPreconditions = true;
+    return;
+  }
+
+  try {
+    // check if coverage store exists
+    const covStoreObject = await grc.datastores.getCoverageStore(ws, covStore);
+
+    if (covStoreObject) {
+      log("Datastore already exists!")
+      return;
+    }
+    // code originally from Sauber project
+    // cf. https://github.com/meggsimum/sauber-sdi-docker/blob/master/geoserver_publisher/index.js
+    log('CoverageStore', covStore, 'does not exist. Try to create it ...');
+
+    ////////////////////////////////
+    ///// indexer.properties ///////
+    ////////////////////////////////
+
+    log('... extending indexer file');
+
+    const indexerFile = process.cwd() + '/gs-img-mosaic-tpl/indexer.properties.tpl';
+    const indexerFileCopy = process.cwd() + '/gs-img-mosaic-tpl/indexer.properties';
+    // copy indexer template so we can modify
+    await fsPromises.copyFile(indexerFile, indexerFileCopy);
+    log(`${indexerFile} was copied to ${indexerFileCopy}`);
+
+    ////////////////////////////////
+    ///// datastore.properties /////
+    ////////////////////////////////
+
+    const dataStoreTemplateFile = process.cwd() + '/gs-img-mosaic-tpl/datastore.properties.tpl';
+    const dataStoreFile = process.cwd() + '/gs-img-mosaic-tpl/datastore.properties';
+
+    log('... replacing dataStore file');
+    const readData = await fsPromises.readFile(dataStoreTemplateFile, 'utf8');
+
+    const adaptedContent = readData.replace(/__DATABASE_PASSWORD__/g, pgPassword);
+    log({ adaptedContent })
+    await fsPromises.writeFile(dataStoreFile, adaptedContent);
+    log('... DONE Replacing datastore file');
+
+    // zip image mosaic properties config files
+    const fileToZip = [
+      'gs-img-mosaic-tpl/indexer.properties',
+      'gs-img-mosaic-tpl/datastore.properties',
+      'gs-img-mosaic-tpl/timeregex.properties'
+    ];
+    const zipPath = '/tmp/init.zip';
+
+    await execShellCommand('zip -j ' + zipPath + ' ' + fileToZip.join(' '));
+
+    await grc.datastores.createImageMosaicStore(ws, covStore, zipPath);
+
+    log('... CoverageStore', covStore, 'created');
+
+  } catch (error) {
+    log(error);
+    throw 'Could not create CoverageStore.';
+  }
+
+  workerJob.status = 'success';
+  workerJob.outputs = [];
```

Comment:
could return the coverage store name, so that following process could use that return value